File: hash/349.两个数组的交集.go
/*
 * @lc app=leetcode.cn id=349 lang=golang
 *
 * [349] 两个数组的交集
 */

// @lc code=start
func intersection(nums1 []int, nums2 []int) []int {
    tempMap := make(map[int]int, 0)
	res := make([]int, 0)
	for _, v := range nums1 {
		if _, ok := tempMap[v]; !ok {
			tempMap[v] = 1
		}
	}
	for _, v := range nums2 {
		if _, ok := tempMap[v]; ok {
			res = append(res, v)
			delete(tempMap, v)
		}
	}
	return res
}
// @lc code=end

File: hash/349.两个数组的交集.js
/*
 * @lc app=leetcode.cn id=349 lang=javascript
 *
 * [349] 两个数组的交集
 */

// @lc code=start
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    let tempSet = new Set(nums1)
    const resSet = new Set()

    for (let i = 0; i < nums2.length; i++) {
        if (tempSet.has(nums2[i])) {
            resSet.add(nums2[i])
            tempSet.delete(nums2[i])
        }
    }

    return Array.from(resSet )
};
// @lc code=end

